### ISSUE TITLE
What is the difference between our Accuracy metric and the built-in version?

### ISSUE THREAD

**Issue by samedii** [codewell/ml-workflow#4]
Ignite's version: https://pytorch.org/ignite/metrics.html#ignite.metrics.Accuracy

@FelixAbrahamsson I'm guessing you had some issues with the default, can you fill us in?

**Comment by FelixAbrahamsson** [comment]
I can't quite remember, but it was either something about wanting to pass around dict objects which the ignite version didn't support at the time, or something about the required shapes for multiclass that I didn't like. However, the current version (0.3.0) of ignite does support dictionaries, but they must have the keys `y_pred` and `y` otherwise you get an error, so the implementations in `workflow.ignite.metrics` do not work anymore.

I just tried using the current versions of the ignite built-in Accuracy and ConfusionMatrix, and I find the Accuracy implementation to be working just fine for both binary and multiclass. I'm not happy with the way their ConfusionMatrix implementation works for the binary case though. Normally in torch the input to the bce loss has shapes (N,) both for the targets and the predictions, and both are type float. This works fine with the sklearn implementation of confusion matrix, for example
```
>>> from sklearn.metrics import confusion_matrix
>>> targets = torch.tensor([1., 0., 0., 1., 0.])
>>> preds = torch.tensor([0., 1., 1., 1., 0.])
>>> confusion_matrix(targets, preds, labels=[0, 1])
array([[1, 2],
       [1, 1]])
```
But to use the ignite version you need to one-hot the predictions and you also need both predictions and targets to be integer type  for it to work, like this
```
output_transform=lambda output: dict(
    y_pred=torch.nn.functional.one_hot(
        (torch.sigmoid(output['predictions']) > 0.5).long()
    ), num_classes=2),
    y=output['targets'].long()
)
```
which I don't like but I can deal with it.
Another thing, SoftAccuracy is basically just 1-mean_absolute_error, so it feels a bit unnecessary. Maybe we just throw out the whole `workflow.ignite.metrics` package for now?

**Comment by samedii** [comment]
@FelixAbrahamsson Is your implementation a "SoftConfusionMatrix" then?

They are quite similar indeed:
1 - mean(abs(target - prediction)) vs. (1 - abs(target - prediction))/len(target)

I think it's up to you. I haven't gotten far enough to get a feel for this yet.

**Comment by FelixAbrahamsson** [comment]
Nope, my confusion matrix implementation is a normal confusion matrix, but it accepts input in the binary case on the same form as what is expected by `torch.nn.BCEWithLogitsLoss`, which is what we usually train with.

I will make a PR to remove those files then :) Would rather keep the library as non-bloated as possible!
